### PR TITLE
tidb-binlog: add trouble shooting and error handling docs

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? 

Add troubleshooting documents for TiDB Binlog.

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/2037

### Which version does your change affect? 

dev, v2.1, v3.0, v3.1

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
LGTM

File: dev/TOC.md [hunk 1]
```diff
@@ -300,6 +300,9 @@
     - [Upgrade](/dev/reference/tidb-binlog/upgrade.md)
     - [Reparo](/dev/reference/tidb-binlog/reparo.md)
     - [Binlog Slave Client](/dev/reference/tidb-binlog/binlog-slave-client.md)
+    + Troubleshoot
+      - [TiDB Binlog Troubleshooting](/dev/reference/tidb-binlog/troubleshoot/binlog.md)
```
Comment (lilin90) [inline dev/TOC.md]:
```suggestion
      - [Troubleshooting](/dev/reference/tidb-binlog/troubleshoot/binlog.md)
```

File: dev/reference/tidb-binlog/troubleshoot/binlog.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+aliases: ['/docs/dev/how-to/troubleshoot/tidb-binlog/']
+---
+
+# TiDB Binlog Troubleshooting
+
+This document describes how to troubleshoot TiDB Binlog to find the problem.
+
+If you encounter errors while running TiDB Binlog, take the following steps to troubleshoot:
+
+1. Check whether each monitoring metrics is normal or not. Refer to [TiDB Binlog Monitoring](/dev/reference/tidb-binlog/monitor.md) for details.
```
Comment (lilin90) [inline dev/reference/tidb-binlog/troubleshoot/binlog.md]:
```suggestion
1. Check whether each monitoring metric is normal or not. Refer to [TiDB Binlog Monitoring](/dev/reference/tidb-binlog/monitor.md) for details.
```

File: dev/reference/tidb-binlog/troubleshoot/binlog.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+2. Use the [binlogctl tool](/dev/reference/tidb-binlog/maintain.md#binlogctl-guide) to check whether the state of each Pump or Drainer node is normal or not.
+
+3. Check whether `ERROR` or `WARN` exists in the Pump log or Drainer log.
+
+After finding out the problem by the above steps, refer to [FAQ](/dev/reference/tidb-binlog/faq.md) and [TiDB Binlog Error Handling](/dev/reference/tidb-binlog/troubleshoot/error-handling.md) for the solution. If you fail to find the solution or the solution provided does not help, submit an [issue](https://github.com/pingcap/tidb-binlog/issues) or contact our technical staff for help.
```
Comment (lilin90) [inline dev/reference/tidb-binlog/troubleshoot/binlog.md]: How to "contact our technical staff"? Please make it clear or direct remove it.

File: dev/reference/tidb-binlog/troubleshoot/error-handling.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+category: reference
+---
+
+# TiDB Binlog Error Handling
+
+This document introduces common errors that you might encounter and solutions to these errors when you use TiDB Binlog.
+
+## `kafka server: Message was too large, server rejected it to avoid allocation error` is returned when Drainer replicates data to Kafka
+
+Cause: Executing a large transaction in TiDB generates binlog data of large size, which might exceed Kafka's limit on message size.
```
Comment (lilin90) [inline dev/reference/tidb-binlog/troubleshoot/error-handling.md]:
```suggestion
Cause: Executing a large transaction in TiDB generates binlog data of a large size, which might exceed Kafka's limit on the message size.
```

**Pull request merged**